clang/lib/Parse/ParseStmt.cpp

```diff
@@ -2636,13 +2637,32 @@ WhereClauseFact *Parser::ParseWhereClauseFact() {
 
   // Parse an equality expression.
   SourceLocation ExprLoc = Tok.getLocation();
-  ExprResult ExprRes = Actions.CorrectDelayedTyposInExpr(ParseExpression());
+
+  // ParseExpression parses expressions including top-level commas. So
+  // declarations like the following are not parsed correctly.
+  // int a _Where a == 1, b;
+  // If fails because it parses "a == 1, b" as one expression.
+
+  // ParseAssignmentExpression parses an expression not including top-level
+  // commas. So the above declaration is parsed correctly. This is also useful
+  // for parsing multiple comma-separated declarations each having its own
+  // where clause as follows:
+  // int a _Where a < 1, b _Where b > 1, c, d, e _Where e == 1;
+
+  ExprResult ExprRes = ParseAssignmentExpression();
```

Comment:
There is example code where `Actions.CorrectDelayedTyposInExpr` is called on the return value of  `ParseAssignmentExpression`. See for example, lines 7293 and 7294 in file clang/lib/Parse/ParseDecl.cpp. Do you want to consider doing the same?